README.md

```diff
@@ -1,3 +1,59 @@
 # liftbridge-grpc
 
 Protobuf definitions for the [Liftbridge](https://github.com/liftbridge-io/liftbridge) gRPC API.
+
+## NATS Message Envelope
+
+Every Liftbridge message that is sent over NATS should be sent with the following
+envelope header:
+
+```
+0               8               16              24              32
+├───────────────┴───────────────┴───────────────┴───────────────┤
+│                          Magic Number                         │
+├───────────────┬───────────────┬───────────────┬───────────────┤
+│    Version    │   HeaderLen   │     Flags     │    MsgType    │
+├───────────────┴───────────────┴───────────────┴───────────────┤
+│                       CRC-32C (optional)                      │
+└───────────────────────────────────────────────────────────────┘
+```
+
+
+#### Magic number [4 bytes]
+
+The Liftbridge magic number is `B9 0E 43 B4`. This was chosen by random but deliberately
+restricted to invalid UTF-8 to reduce the chance of a collision. This was also verified
+to not match known file sigatures.
+
+#### Version [1 byte]
+
+The version byte allows for future protocol upgrades. This should only be bumped if the
+envelope format changes or if the message encoding changes in a non-backwards-compatible
+way. Adding fields to the messages should not require a version bump.
+
+#### HeaderLen [1 byte]
+
+The header length is the offset of the payload. This is included primarily for safety.
+
+#### Flags [1 byte]
+
+The flag bits are defined as follows:
+
+| Bit | Description     |
+| --- | --------------- |
+| 0   | CRC-32C enabled |
+
+#### MsgType [1 byte]
+
+This is the Liftbridge-specific message type
+
+| MsgType | Description |
+| ------- | ----------- |
+| 0       | Publish     |
+| 1       | Replication |
```

Comment:
I thought we were considering using the same envelope for the replication messges as well, in which case we need to know which message type to decode the payload with.

Reply:
How would we be handling decoding differently between publishes vs. replication? In either case we should know if it's a publish or a replicated message based on the context in which the message is received, right?

Reply:
Yes, I suppose maybe I'm just being overly cautious. If the user happens to send liftbridge-encoded publishes over the replication subject, it could be catastrophic. But maybe we can just choose a different magic number for the replication header and call it a day?  I'm fine with that, and we can keep this byte reserved for better use cases down the road.

Reply:
Different magic number makes sense to me, but I can be convinced otherwise.